internal/powerplatform/provider.go

```diff
@@ -258,6 +276,72 @@ func (p *PowerPlatformProvider) Configure(ctx context.Context, req provider.Conf
+		p.Config.Urls.PowerPlatformScope = constants.PUBLIC_POWERPLATFORM_API_SCOPE
+		p.Config.Cloud = azcloud.AzurePublic
+	case "gcc":
+		p.Config.Urls.BapiUrl = constants.USGOV_BAPI_DOMAIN
+		p.Config.Urls.PowerAppsUrl = constants.USGOV_POWERAPPS_API_DOMAIN
+		p.Config.Urls.PowerAppsScope = constants.USGOV_POWERAPPS_SCOPE
+		p.Config.Urls.PowerPlatformUrl = constants.USGOV_POWERPLATFORM_API_DOMAIN
+		p.Config.Urls.PowerPlatformScope = constants.USGOV_POWERPLATFORM_API_SCOPE
+		p.Config.Cloud = azcloud.AzurePublic //GCC uses public cloud for authentication
+	case "gcchigh":
+		p.Config.Urls.BapiUrl = constants.USGOVHIGH_BAPI_DOMAIN
+		p.Config.Urls.PowerAppsUrl = constants.USGOVHIGH_POWERAPPS_API_DOMAIN
+		p.Config.Urls.PowerAppsScope = constants.USGOVHIGH_POWERAPPS_SCOPE
+		p.Config.Urls.PowerPlatformUrl = constants.USGOVHIGH_POWERPLATFORM_API_DOMAIN
+		p.Config.Urls.PowerPlatformScope = constants.USGOVHIGH_POWERPLATFORM_API_SCOPE
+		p.Config.Cloud = azcloud.AzureGovernment
+	case "dod":
+		p.Config.Urls.BapiUrl = constants.USDOD_BAPI_DOMAIN
+		p.Config.Urls.PowerAppsUrl = constants.USDOD_POWERAPPS_API_DOMAIN
+		p.Config.Urls.PowerAppsScope = constants.USDOD_POWERAPPS_SCOPE
+		p.Config.Urls.PowerPlatformUrl = constants.USDOD_POWERPLATFORM_API_DOMAIN
+		p.Config.Urls.PowerPlatformScope = constants.USDOD_POWERPLATFORM_API_SCOPE
+		p.Config.Cloud = azcloud.AzureGovernment
+	case "china":
+		p.Config.Urls.BapiUrl = constants.CHINA_BAPI_DOMAIN
+		p.Config.Urls.PowerAppsUrl = constants.CHINA_POWERAPPS_API_DOMAIN
+		p.Config.Urls.PowerAppsScope = constants.CHINA_POWERAPPS_SCOPE
+		p.Config.Urls.PowerPlatformUrl = constants.CHINA_POWERPLATFORM_API_DOMAIN
+		p.Config.Urls.PowerPlatformScope = constants.CHINA_POWERPLATFORM_API_SCOPE
+		p.Config.Cloud = azcloud.AzureChina
+	case "ex":
+		p.Config.Urls.BapiUrl = constants.EX_BAPI_DOMAIN
+		p.Config.Urls.PowerAppsUrl = constants.EX_POWERAPPS_API_DOMAIN
+		p.Config.Urls.PowerAppsScope = constants.EX_POWERAPPS_SCOPE
+		p.Config.Urls.PowerPlatformUrl = constants.EX_POWERPLATFORM_API_DOMAIN
+		p.Config.Urls.PowerPlatformScope = constants.EX_POWERPLATFORM_API_SCOPE
+		p.Config.Cloud = azcloud.Configuration{
+			ActiveDirectoryAuthorityHost: constants.EX_AUTHORITY_HOST,
+			Services:                     map[azcloud.ServiceName]azcloud.ServiceConfiguration{},
+		}
+	case "rx":
+		p.Config.Urls.BapiUrl = constants.RX_BAPI_DOMAIN
+		p.Config.Urls.PowerAppsUrl = constants.RX_POWERAPPS_API_DOMAIN
+		p.Config.Urls.PowerAppsScope = constants.RX_POWERAPPS_SCOPE
+		p.Config.Urls.PowerPlatformUrl = constants.RX_POWERPLATFORM_API_DOMAIN
+		p.Config.Urls.PowerPlatformScope = constants.RX_POWERPLATFORM_API_SCOPE
+		p.Config.Cloud = azcloud.Configuration{
+			ActiveDirectoryAuthorityHost: constants.RX_AUTHORITY_HOST,
+			Services:                     map[azcloud.ServiceName]azcloud.ServiceConfiguration{},
+		}
+	default:
+		resp.Diagnostics.AddAttributeError(
```

Comment:
so no cloud param throws an error? Shouldn't the "public" value be default is no param is specified? How does it work with az login then if you just need public cloud?

Reply:
We default to "public" above if you don't pass anything.  If you pass something else that we don't recognize then we error.